components/renku_data_services/k8s/clients.py

```diff
@@ -75,7 +76,13 @@ class DummyCoreClient(K8sCoreClientInterface):
 
     def __init__(self, quotas: Dict[str, client.V1ResourceQuota]):
         self.quotas = quotas
-        self._lock = Lock()
+        self.__lock: LockType | None = None
+
+    @property
+    def _lock(self):
+        if not self.__lock:
+            self.__lock = Lock()
+        return self.__lock
```

Comment:
The problem is that if it starts before Sanic starts then Sanic crashes. With the property we initialize the multiprocessing Lock after Sanic has started and it does not crash the Sanic server.

With this change it is much easier/simpler to run Sanic locally. Otherwise it worked only with --single-process before.

Reply:
in that case there should be a comment describing why this is here